skimage/__init__.py

```diff
@@ -1,68 +1,11 @@
 """Image Processing for Python
 
-``scikit-image`` (a.k.a. ``skimage``) is a collection of algorithms for image
+scikit-image (a.k.a. ``skimage``) is a collection of algorithms for image
 processing and computer vision.
 
 The main package of ``skimage`` only provides a few utilities for converting
-between image data types; for most features, you need to import one of the
-following subpackages:
-
-Subpackages
------------
-color
-    Color space conversion.
-data
-    Test images and example data.
-draw
-    Drawing primitives (lines, text, etc.) that operate on NumPy arrays.
-exposure
-    Image intensity adjustment, e.g., histogram equalization, etc.
-feature
-    Feature detection and extraction, e.g., texture analysis corners, etc.
-filters
-    Sharpening, edge finding, rank filters, thresholding, etc.
-graph
-    Graph-theoretic operations, e.g., shortest paths.
-io
-    Reading, saving, and displaying images and video.
-measure
-    Measurement of image properties, e.g., region properties and contours.
-metrics
-    Metrics corresponding to images, e.g. distance metrics, similarity, etc.
-morphology
-    Morphological operations, e.g., opening or skeletonization.
-restoration
-    Restoration algorithms, e.g., deconvolution algorithms, denoising, etc.
-segmentation
-    Partitioning an image into multiple regions.
-transform
-    Geometric and other transforms, e.g., rotation or the Radon transform.
-util
-    Generic utilities.
```

Comment:
Why remove this list? I suspect the goal is to avoid duplication of docstrings, but I think discoverability is important here.

Reply:
Yes, the goal is avoiding duplication with regards to these getting out of sync, requiring manual maintenance, and because keeping these also lead to [duplication in our docs](https://scikit-image.org/docs/stable/api/skimage.html). E.g. it is outdated / misleading right now, `registration` is missing from this list.

And since `help(skimage)`, `dir(skimage)`, and auto-completion tools will all display the available subpackages, I didn't really saw the benefit. It's also not a standard thing that's done in our ecosystem (I think).

I agree with your argument but when weighting with the others, I think removal is the right call. But we can revert if you feel different.

Reply:
The first one I checked was SciPy:

```
Subpackages
-----------
Using any of these subpackages requires an explicit import. For example,
``import scipy.cluster``.

::

 cluster                      --- Vector Quantization / Kmeans
 constants                    --- Physical and mathematical constants and units
 datasets                     --- Dataset methods
 fft                          --- Discrete Fourier transforms
 fftpack                      --- Legacy discrete Fourier transforms
 integrate                    --- Integration routines
 interpolate                  --- Interpolation Tools
 io                           --- Data input and output
 linalg                       --- Linear algebra routines
 misc                         --- Utilities that don't have another home.
 ndimage                      --- N-D image package
 odr                          --- Orthogonal Distance Regression
 optimize                     --- Optimization Tools
 signal                       --- Signal Processing Tools
 sparse                       --- Sparse Matrices
 spatial                      --- Spatial data structures and algorithms
 special                      --- Special functions
 stats                        --- Statistical Functions
```

For people interactively exploring at the prompt, there is no other way to concisely get a list of submodules + descriptions, AFAIK?

Reply:
Ah, they curate those by hand: https://github.com/scipy/scipy/blob/main/doc/source/reference/main_namespace.rst

Then let's revert this with added `registration` and updated summaries. Any ideas on avoiding the duplication and checking that the lists are still correct?

Reply:
There are ways, but they inject complication; let's just try and remember :)

Reply:
Done in https://github.com/scikit-image/scikit-image/pull/6892/commits/6d7a770d70c4c95b31761b6f44e014876ad13f0e. `registration` was still missing summary in its init file as well so you might want to check what I added there.

Reply:
We could say "Image alignment algorithms" to make it more obvious what it does; but also fine as-is. Thanks, Lars.